### ISSUE TITLE
Correctly hint TLS when the target pod has skip-inbound-ports

### ISSUE THREAD

**Issue by adleong** [linkerd/linkerd2#6937]
When the target pod uses skip-inbound-ports, the destination controller should not hint that the target has identity.  Hinting that the target has identity will cause the source proxy to initiate mesh TLS that the target pod cannot terminate.

### Discussed in https://github.com/linkerd/linkerd2/discussions/6915

<div type='discussions-op-text'>

<sup>Originally posted by **lynshi** September 17, 2021</sup>
We are currently using linkerd2 2.8.1 in a setup where some Pods have their own TLS with self-signed certificates. In 2.8.1, if Pod A has self-signed TLS on port 1000, with Service A linking to `containerPort` 1000 from 443, then we add 1000 to `--skip-inbound-ports` and everything works fine.

In linkerd 2.9 and 2.10, we are seeing the following errors in `linkerd-proxy`:
```
[   975.821558s]  WARN ThreadId(01) server{orig_dst=<IP address>}: rustls::session: Sending fatal alert BadCertificate
```
and we cannot connect to Pod A via Service A. To resolve this issue, we have to also add 443 to `--skip-outbound-ports` on the connecting Pod. Does this mean that (after linkerd 2.8) the proxy on the Pod with the outgoing connection is validating the TLS certificate, so the proxy must be disabled on both ends?</div>

**Comment by adleong** [comment]
It looks like this was fixed in https://github.com/linkerd/linkerd2/pull/6022 after 2.10 was released.  This fix is available in recent edge releases and will be included in stable-2.11.0

**Comment by adleong** [comment]
Duplicate of https://github.com/linkerd/linkerd2/issues/5995
